src/rate_limiter.rs

```diff
@@ -167,3 +168,54 @@ impl Default for RateLimiter {
         Self::new()
     }
 }
+
+#[cfg(feature = "use-mock-crust")]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use messages::Request;
+    use rand;
+    use types::MessageId;
+
+    fn sleep_millis(time: u64) {
```

Comment:
maybe get rid of this fn and just use the FakeClock direct where needed. Prolly same for the following fn too